### PR TITLE
Set types in arrays

### PR DESCRIPTION
**Pull request by szepeviktor**
as arrays are not really types but containers

### CODE REVIEW

File: src/HttpPartialResponse.php [hunk 1]
```diff
@@ -21,12 +21,13 @@ final class HttpPartialResponse implements PartialResponse
...
     /**
-     * @return HttpPartialResponse[]
+     * @param list<string> $lines
+     * @return list<HttpPartialResponse>
      */
     public static function parseLines(array $lines): array
     {
         $partials = [];
-        while (count($lines) > 0) {
+        while ($lines !== []) {
```
Comment (szepeviktor) [inline src/HttpPartialResponse.php:31]: This is 1 line of explicit programming :)

**Comment by szepeviktor** [comment]
@phpstan is a bit better software, has a nicer birth place and brighter future

File: src/HttpPartialResponse.php [hunk 2]
```diff
@@ -21,6 +21,7 @@ final class HttpPartialResponse implements PartialResponse
     private Headers $headers;
 
     /**
+     * @param HttpPartialResponse[] $lines
```
Comment (mnavarrocarter) [inline src/HttpPartialResponse.php]: The `$lines` param is actually `string[]`

**Review by mnavarrocarter — CHANGES_REQUESTED** [review]
Hi @szepeviktor thanks for this: improves type information a lot which I always welcome. 

If you can just fix the little comment I left I will definitely merge this.

Thanks!

File: src/HttpPartialResponse.php (hunk 2 above)
Comment (szepeviktor) [reply src/HttpPartialResponse.php]: Oh no! It proves I'm no developer (but a janitor)

Fixing the fix .... :)

Comment (szepeviktor) [reply src/HttpPartialResponse.php]: Oh no Again :)

It was already fixed in the second commit.

Comment (mnavarrocarter) [reply src/HttpPartialResponse.php]: Oh sorry, I probably wrote the comment before looking at the other commit.

**Pull request merged**